external/llvm-project/mlir/lib/Conversion/GPUCommon/OpToFuncCallLowering.h

```diff
@@ -78,7 +78,15 @@ struct OpToFuncCallLowering : public ConvertOpToLLVMPattern<SourceOp> {
 private:
   Value maybeCast(Value operand, PatternRewriter &rewriter) const {
     Type type = operand.getType();
-    if (!isa<Float16Type>(type))
+    if (isa<ShapedType>(type)) {
```

Comment:
I think the condition might want to be `!isa<FloatType>(type))`?

Also, no need for braces here, I think